GitHub.Collectors/Processor/PointProcessor.cs

```diff
@@ -0,0 +1,86 @@
+﻿using Microsoft.CloudMine.Core.Collectors.Authentication;
+using Microsoft.CloudMine.Core.Collectors.Cache;
+using Microsoft.CloudMine.Core.Collectors.Collector;
+using Microsoft.CloudMine.Core.Collectors.IO;
+using Microsoft.CloudMine.Core.Collectors.Telemetry;
+using Microsoft.CloudMine.GitHub.Collectors.Cache;
+using Microsoft.CloudMine.GitHub.Collectors.Collector;
+using Microsoft.CloudMine.GitHub.Collectors.Web;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Microsoft.CloudMine.GitHub.Collectors.Processor
+{
+    public class PointProcessor
+    {
+        private readonly CollectorBase<GitHubCollectionNode> collector;
+
+        private readonly List<IRecordWriter> recordWriters;
+        private readonly ICache<PointCollectorTableEntity> cache;
+        private readonly ITelemetryClient telemetryClient;
+        private readonly string apiDomain;
+
+        public PointProcessor(IAuthentication authentication,
+                              List<IRecordWriter> recordWriters,
+                              GitHubHttpClient httpClient,
+                              ICache<PointCollectorTableEntity> cache,
+                              ITelemetryClient telemetryClient,
+                              string apiDomain)
+        {
+            this.collector = new GitHubCollector(httpClient, authentication, telemetryClient, recordWriters);
+            this.recordWriters = recordWriters;
+            this.cache = cache;
+            this.telemetryClient = telemetryClient;
+            this.apiDomain = apiDomain;
+        }
+
+        public async Task ProcessAsync(PointCollectorInput input)
+        {
+            Dictionary<string, JToken> additionalMetadata = new Dictionary<string, JToken>();
+            JObject context = input.Context;
+            if (context != null)
+            {
+                foreach (KeyValuePair<string, JToken> contextElement in context)
+                {
+                    additionalMetadata.Add(contextElement.Key, contextElement.Value);
+                }
+            }
+    
+            GitHubCollectionNode collectionNode = new GitHubCollectionNode()
+            {
+                RecordType = input.RecordType,
+                ApiName = input.ApiName,
+                GetInitialUrl = metadata => input.Url,
+                AdditionalMetadata = additionalMetadata,
+            };
+
+            foreach (IRecordWriter recordWriter in this.recordWriters)
+            {
+                await recordWriter.NewOutputAsync(input.RecordType).ConfigureAwait(false);
+            }
```

Comment:
Do we really need  a new output just to change the output name wrt record type? I think it is fine for this to stay as Point_<blah>, what do you think?